crates/net/eth-wire/src/types/message.rs

```diff
@@ -301,13 +301,13 @@ impl<T: Encodable> EncodableExt for Vec<T> {
             header.payload_length += item.length();
         }
         header.encode(&mut buffer);
-
         for item in self {
             item.encode(&mut buffer);
             if buffer.len() > limit {
                 return Err(alloy_rlp::Error::Custom("Size limit exceeded"));
```

Comment:
asides this, the two methods are identical (`let current_len = buffer.len();` in the other trait method can be moved down to the line after the conditional statement). can we instead have a third method `encode_until_limit` with this code which returns `(Vec<u8>, bool)`. then `encode_max` will be smgth like
```rust
 fn encode_max(
        &self,
        approx: usize,
        limit: usize,
    ) -> alloy_rlp::Result<Vec<u8>, alloy_rlp::Error> {
    let (mut buf, limit_exceeded) = self.encode_until_limit(aprox, limit);
    
    if limit_exceeded {
        return Err(alloy_rlp::Error::Custom("Size limit exceeded"));
    }
    
    buf.shrink_to_fit();
    
    Ok(buf)
 }

```
and encode_truncate can be simplified too then. that way we don't have duplicate code, and it's easier to see how the methods' behaviour is different.